tests/rptest/remote_scripts/topic_operations.py

```diff
@@ -285,21 +287,69 @@ def _avoidable_char_sequences(new_name):
         timings['end_time_s'] = time.time()
         return (topics, timings)
 
+    def delete_many_topics(self, topic_name_prefix):
+        """Function deletes all topics based on the prefix
+
+        Args:
+            topic_name_prefix (list): topics list
+
+        Returns:
+            list[str], dict: topic names as list and timings dict
+        """
+        def _delete_topic_batch(topic_item_list):
+            # Send whole batch to kafka client
+            r = kclient.delete_topics(topic_item_list,
+                                      timeout_ms=self.create_topic_timeout_ms)
+            # Filter topic errors if any
+            _errors = [
+                e for e in getattr(r, 'topic_error_codes', []) if e[1] != 0
+            ]
+            # Transpose errors so names appear on 0, error code in 1
+            # and Class in 2
+            _errors = numpy.array(_errors).transpose()
+            # Fill time and check errors
+            for item in topic_item_list:
+                # if there is an error, create key
+                if len(_errors) > 0 and item['name'] in _errors[0]:
+                    idx = _errors[0].index(item['name'])
+                    item['topic_errors'] = (_errors[1][idx], _errors[2][idx])
+            return topic_item_list
+
+        kclient = self.make_superuser_client()
+        # list topics with prefix
+        all_topics = kclient.list_topics()
+        topics_to_delete = []
+        for topic_name in all_topics:
+            if topic_name.startswith(topic_name_prefix):
+                topics_to_delete.append(topic_name)
+
+        # delete them
+        delete_time_start_s = time.time()
+        _delete_topic_batch(topics_to_delete)
+        batch_deletion_time_s = time.time() - delete_time_start_s
+
+        # Peek creation timings
+        timings = {
+            "count_deleted": len(topics_to_delete),
+            "deletion_time": batch_deletion_time_s
+        }
+        return (topics_to_delete, timings)
+
 
 COMMAND_CREATE = 'create'
 COMMAND_DELETE = 'delete'
 commands = [COMMAND_CREATE, COMMAND_DELETE]
 
 
 def main(args):
-
+    errorlevel = 0
     tm = TopicSwarm(args.brokers,
                     args.workers,
                     args.issilent,
                     logger=setup_logger())
 
-    if args.command == COMMAND_CREATE:
-        try:
+    try:
+        if args.command == COMMAND_CREATE:
```

Comment:
This would be a great place to use `match`/`case` as mentioned on Slack, but no pressure or anything. :)